addOns/ascanrulesAlpha/src/main/java/org/zaproxy/zap/extension/ascanrulesAlpha/SpringActuatorScanRule.java

```diff
@@ -0,0 +1,193 @@
+/*
+ * Zed Attack Proxy (ZAP) and its related class files.
+ *
+ * ZAP is an HTTP/HTTPS proxy for assessing web application security.
+ *
+ * Copyright 2021 The ZAP Development Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.zaproxy.zap.extension.ascanrulesAlpha;
+
+import java.io.IOException;
+import java.util.regex.Pattern;
+import org.apache.commons.httpclient.URI;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.parosproxy.paros.Constant;
+import org.parosproxy.paros.core.scanner.AbstractHostPlugin;
+import org.parosproxy.paros.core.scanner.Alert;
+import org.parosproxy.paros.core.scanner.Category;
+import org.parosproxy.paros.network.HttpHeader;
+import org.parosproxy.paros.network.HttpMessage;
+import org.parosproxy.paros.network.HttpRequestHeader;
+import org.zaproxy.zap.model.Tech;
+import org.zaproxy.zap.model.TechSet;
+
+/**
+ * An active scan rule to test for Spring Actuators revealing too much info. Health might be the
+ * most commonly enabled but if health is enabled, so might be Thread Dump or Heap Dump which could
+ * be serious https://docs.spring.io/spring-boot/docs/current/actuator-api/htmlsingle/#overview
+ *
+ * @author sgerlach
+ */
+public class SpringActuatorScanRule extends AbstractHostPlugin {
+
+    private static final String MESSAGE_PREFIX = "ascanalpha.springactuator.";
+    private static final int PLUGIN_ID = 40042;
+    private static final Logger LOG = LogManager.getLogger(SpringActuatorScanRule.class);
+    private static final Pattern CONTENT_TYPE =
+            Pattern.compile(
+                    "application\\/vnd\\.spring-boot\\.actuator\\.v[0-9]\\+json|application\\/json",
+                    Pattern.MULTILINE);
+    private static final Pattern JSON_PAYLOAD = Pattern.compile("\\{.*\\:.*\\}", Pattern.MULTILINE);
+
+    @Override
+    public boolean targets(TechSet technologies) {
+        // TODO change this to Spring, but need to extend the Tech in Core ZAP first
```

Comment:
If you'd like to get Spring into the core tech then now would be the time to do it ;)

Reply:
would love to talk about how to get frameworks in there since they are technically subsets of languages.... perhaps not an easy add...

Reply:
Shouldn't be too big a deal we have generic DB, and specific brands or flavours. Should be similar.

Reply:
The todo comment can be addressed now, apply the changes like https://github.com/zaproxy/zap-extensions/pull/3182/files#diff-44a98f40cdfd21bf5c0d576508dc9fc324b5da7b7679cf3ed72df456f6d1c11c and use the ZAP version `zap-2.11.0-20210929.165234-4`.